### DISCUSSION TITLE
Extend syntax highlighting and completions for string types

### DISCUSSION THREAD

**Discussion by cdce8p** [microsoft/pylance-release#2631] [Enhancement]
I think it's awesome that Pylance supports syntax highlighting and completions in string type annotations.

<img width="283" alt="Screen Shot 2022-01-26 at 14 49 10" src="https://user-images.githubusercontent.com/30130371/151174719-9dafe025-9995-41c1-91ec-d2b081f3b88b.png">

Would it be possible to extend that? Below `cast` and `TypeVar` as example. `NewType` is also similar.

<img width="453" alt="Screen Shot 2022-01-26 at 14 51 23" src="https://user-images.githubusercontent.com/30130371/151175108-433a1358-d81a-4525-b39d-85afd4ab0ad3.png">

Lastly, type aliases. Variables explicitly annotated with `TypeAlias` should be strait forward. Some support for it seems to be implemented already. Implicit aliases are probably a bit more difficult. Although cases which can safely be inferred as alias might still be possible.

<img width="460" alt="Screen Shot 2022-01-26 at 15 19 41" src="https://user-images.githubusercontent.com/30130371/151179819-7f6b3a8e-6058-4051-a740-21ce687562b4.png">

<details>
<summary>Example code</summary>

```py
from typing import Callable, NewType, TypeAlias, TypeVar, cast

T = TypeVar("T", bound="SomeClass")
T2 = TypeVar("T2", "SomeClass", int)

def func(var) -> "SomeClass":
    return cast("SomeClass", var)

X = NewType("X", "SomeClass")

Alias1: TypeAlias = "Callable[[], SomeClass]"
Alias2 = Callable[[], "SomeClass"]

class SomeClass: ...
```
</details>

**Comment by judej** [comment]
Moving this issue to discussion as an enhancement request for comments and upvotes.

**Comment by Viicos** [comment]
+1 especially because using forward references in this context is pretty common when supporting Python < 3.10 in libraries. Using `from __future__ import annotations` isn't an option in these cases as the arguments/type aliases are evaluated at runtime
